nomad/structs/structs.go

```diff
@@ -316,9 +316,9 @@ type NodeRegisterRequest struct {
 }
 
 // NodeDeregisterRequest is used for Node.Deregister endpoint
-// to deregister a node as being a schedulable entity.
+// to deregister a batch of nodes from being schedulable entities.
 type NodeDeregisterRequest struct {
-	NodeID string
+	NodeIDs []string
```

Comment:
Are there backward compatibility concerns here, specially given that this gets serialized in the fsm?  While upgrading servers, we need to maintain backward compatibility (new servers must process raft log messages from old servers) and forward compatibility (new server leader must produce messages that old servers can process).

I think there is some logic elsewhere we can pattern against to maintain compatibility but don't have references right now - will update review with details.

Reply:
oof, that's a good catch, I hadn't considered serialized compatibility!